port/win/io_win.cc

```diff
@@ -847,29 +873,64 @@ WinWritableImpl::WinWritableImpl(WinFileData* file_data, size_t alignment)
 
 Status WinWritableImpl::AppendImpl(const Slice& data) {
 
-  // Used for buffered access ONLY
-  assert(!file_data_->UseDirectIO());
+  Status s;
+
   assert(data.size() < std::numeric_limits<DWORD>::max());
 
-  Status s;
+  uint64_t written = 0;
 
-  DWORD bytesWritten = 0;
-  if (!WriteFile(file_data_->GetFileHandle(), data.data(),
-    static_cast<DWORD>(data.size()), &bytesWritten, NULL)) {
-    auto lastError = GetLastError();
-    s = IOErrorFromWindowsError(
-      "Failed to WriteFile: " + file_data_->GetName(),
-      lastError);
+  if (file_data_->UseDirectIO()) {
```

Comment:
What changed the `assert(!file_data_->UseDirectIO());` ?
Where are we using AppendImpl with DirectIO enabled